sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala

```diff
@@ -109,13 +109,18 @@ private[sql] class SharedState(
    * A status store to query SQL status/metrics of this Spark application, based on SQL-specific
    * [[org.apache.spark.scheduler.SparkListenerEvent]]s.
    */
-  val statusStore: SQLAppStatusStore = {
-    val kvStore = sparkContext.statusStore.store.asInstanceOf[ElementTrackingStore]
-    val listener = new SQLAppStatusListener(conf, kvStore, live = true)
-    sparkContext.listenerBus.addToStatusQueue(listener)
-    val statusStore = new SQLAppStatusStore(kvStore, Some(listener))
-    sparkContext.ui.foreach(new SQLTab(statusStore, _))
-    statusStore
+  val statusStore: SQLAppStatusStore = SharedState.synchronized {
```

Comment:
`SharedState` is shared across multiple sessions. How can this be created multiple times?

Reply:
code like this:
```
val spark = SparkSession.builder()
.config(sparkConf)
.getOrCreate()

setDefaultSession(null)
setActiveSession(null)

val spark2 = SparkSession.builder()
.config(sparkConf)
.getOrCreate()
```

in `getOrCreate`  function,`new SparkSession(sparkContext, None, None, extensions, options.toMap)`  new SparkSession object will be created,so sharedState will be created multiple times.